File: app/src/main/java/com/zm/audio/java/service/UploadingService.java
package com.zm.audio.java.service;


import android.app.IntentService;
import android.content.Intent;
import android.support.annotation.Nullable;
import android.util.Log;

/**
 * Created by ZhouMeng on 2018/9/11.
 * 上传大文件
 */

public class UploadingService extends IntentService {
    private final static String TAG = "UploadingService";

    /**
     * Creates an IntentService.  Invoked by your subclass's constructor.
     * <p>
     * //     * @param name Used to name the worker thread, important only for debugging.
     */
    public UploadingService() {
        super(TAG);
    }

    @Override
    protected void onHandleIntent(@Nullable Intent intent) {
        Log.d(TAG,"上传操作");
    }
}

File: app/src/main/java/com/zm/audio/kotlin/service/UploadingService.kt
package com.zm.audio.kotlin.service

import android.app.IntentService
import android.content.Intent
import android.util.Log

/**
 * Created by ZhouMeng on 2018/9/13.
 * 开启服务上传文件
 */
class UploadingService : IntentService(TAG) {

    override fun onHandleIntent(intent: Intent?) {
        Log.d(TAG,"上传操作")
    }

    companion object{
        private const val TAG :String = "UploadingService"
    }
}